### ISSUE TITLE
Consider silencing the UnusedVariable violation

### ISSUE THREAD

**Issue by GaryJones** [Automattic/VIP-Coding-Standards#498]
From #449:

> We could also take this opportunity is taken to silence the UnusedVariable violation. It's severity is already dropped down to 1 for `WordPress-VIP-Go`, which is the same as what the bot runs at (though this may change in the future). From a VIP perspective, unused variables are not important.

## What problem would the enhancement address for VIP?

It would reduce the number of violation comments. Although it's not ideal to have values being to variables that are then unused (waste of memory), it's not a severe enough impact to typically worry about.

If clients were keen on this, they could run the VariableAnalyis sniff (or static analysis tools or the inspector plugin for PHPStorm), to find these.

## Describe the solution you'd like

Discuss and consider silencing the UnusedVariable violation.

## What code should be reported as a violation?

...

## What code should *not* be reported as a violation?

...

**Comment by GaryJones** [comment]
One example where this is useful, would be when subsequent usages of a variable have a different name:

```php
$foo_example = 'bar';
...
if ( $fo_example ) {...
```

**Comment by GaryJones** [comment]
No discussion, so punting.

**Comment by rebeccahum** [comment]
This one needs extensive testing of the VariableAnalysis beta sniff as well.

**Comment by rebeccahum** [comment]
I'd like to see `VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable` silenced as well, and the other ones brought down in severity.
